dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
+        # if associated with a registered viewset, use its URL
+        url = getattr(self, '_url', None)
```

Comment:
Yeah, dropping 2) doesn't seem too bad. If you have two similar serializers that need to share code, you can just extend a shared base class into two separate classes, so doesn't seem too burdensome. I don't think I've used 3) so don't have much context on why one would do that, but seems like the right solution should be to either differentiate behaviors in the serializer or just have separate viewsets. So, yeah, 👍 to dropping 2) and 3) from my end.